### PR TITLE
added one test

### PR DESCRIPTION
**Pull request by ashchavinska**

### CODE REVIEW

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java [hunk 1]
```diff
@@ -79,4 +84,39 @@ public CarinaDescriptionPageBase login() {
...
 
+    @Override
+    public boolean isPageOpened() {
+        return nameInputField.isElementPresent() &&
+               passwordInputField.isElementPresent() &&
+               maleRadioBtn.isElementPresent() &&
+               femaleRadioBtn.isElementPresent() &&
+               privacyPolicyCheckbox.isElementPresent() &&
+               loginBtn.isElementPresent();
+    }
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java]: Its bad practice, better when you verify only one specific element on this page
Example: title CARINA

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java [hunk 2]
```diff
@@ -79,4 +84,39 @@ public CarinaDescriptionPageBase login() {
...
+
+    @Override
+    public boolean isPrivacyPolicyChecked() {
+        return privacyPolicyCheckbox.isChecked();
+    }
+
+    @Override
+    public boolean isNamePrinted(){
+        return nameInputField.getSize()!=null;
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java]: verify text which you type before, its will be better
Example: 
public boolean isTypedNamePresent(String name) {
return name.equals(nameInputField.getText);

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java [hunk 3]
```diff
@@ -79,4 +84,39 @@ public CarinaDescriptionPageBase login() {
...
+    }
+
+    @Override
+    public boolean isPasswordPrinted(){
+        return passwordInputField.getSize()!=null;
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java]: see above

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/LoginPageBase.java [hunk 4]
```diff
@@ -24,4 +26,15 @@ public LoginPageBase(WebDriver driver) {
 
 	public abstract CarinaDescriptionPageBase login();
 
+
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/LoginPageBase.java]: Unnecessary space, remove

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 5]
```diff
@@ -3,6 +3,7 @@
 import com.zebrunner.agent.core.annotation.TestLabel;
 import org.apache.commons.lang3.RandomStringUtils;
 import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: remove unused import

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 6]
```diff
@@ -85,4 +86,34 @@ public void testUIElements() {
         Assert.assertTrue(uiElements.isOthersRadioButtonSelected(), "Others radio button was not selected!");
     }
 
+
+
+    @Test(description = "01")
+    @MethodOwner(owner = "ashchavinska")
+    public void verifyLoginPage(){
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: add space between ) and {

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 7]
```diff
@@ -85,4 +86,34 @@ public void testUIElements() {
...
+        String username = "Rachel Green";
+        String password = "1234567890";
+        WelcomePageBase welcomePage = initPage(getDriver(), WelcomePageBase.class);
+        LoginPageBase loginPage = welcomePage.clickNextBtn();
+        Assert.assertTrue(loginPage.isPageOpened(), "Login page isn't opened");
+        //verify fields are presented->they must be (name,password,male/female... already checked previously
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: bad practice, isPageOpened must check only if page opened
if you need verify a lot of fields use one by one(one method for one field)
Example: isNameFieldPresented, ....

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 8]
```diff
@@ -85,4 +86,34 @@ public void testUIElements() {
...
+        Assert.assertFalse((loginPage.isMaleSexChecked() && loginPage.isFemaleSexChecked()), "Sex already checked");
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: better use one assert for one method, its will be more informative when test will be failed

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 9]
```diff
@@ -85,4 +86,34 @@ public void testUIElements() {
...
+        Assert.assertFalse(loginPage.isPrivacyPolicyChecked(), "Privacy policy already checked");
+        loginPage.typeName(username);
+        loginPage.typePassword(password);
+        Assert.assertTrue((loginPage.isNamePrinted() && loginPage.isPasswordPrinted()), "Name field or password field is empty");
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: see above

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 10]
```diff
@@ -85,4 +86,34 @@ public void testUIElements() {
...
+        loginPage.selectFemaleSex();
+        Assert.assertTrue((loginPage.isMaleSexChecked() || loginPage.isFemaleSexChecked()), "Sex isn't checked");
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: see above

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java (hunk 1 above)
Comment (ashchavinska) [reply src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java]: done

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java (hunk 2 above)
Comment (ashchavinska) [reply src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java]: done

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java (hunk 3 above)
Comment (ashchavinska) [reply src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java]: done

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/LoginPageBase.java (hunk 4 above)
Comment (ashchavinska) [reply src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/LoginPageBase.java]: done

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java (hunk 5 above)
Comment (ashchavinska) [reply src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: done

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java (hunk 6 above)
Comment (ashchavinska) [reply src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: done

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java (hunk 7 above)
Comment (ashchavinska) [reply src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: done

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java (hunk 8 above)
Comment (ashchavinska) [reply src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: done

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java (hunk 9 above)
Comment (ashchavinska) [reply src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: done

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java (hunk 10 above)
Comment (ashchavinska) [reply src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: done

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 11]
```diff
@@ -85,4 +86,41 @@ public void testUIElements() {
...
+        SoftAssert softAssert = new SoftAssert();
+        WelcomePageBase welcomePage = initPage(getDriver(), WelcomePageBase.class);
+        LoginPageBase loginPage = welcomePage.clickNextBtn();
+        Assert.assertTrue(loginPage.isPageOpened(), "Login page isn't opened");
+
+        softAssert.assertTrue(loginPage.isNameFieldPresent(), "User name field isn't present");
+        softAssert.assertTrue(loginPage.isPasswordFieldPresent(), "Password field isn't present");
+        softAssert.assertTrue(loginPage.isMaleSexFieldPresent(), "MaleSex field isn't present");
+        softAssert.assertTrue(loginPage.isFemaleSexFieldPresent(), "FemaleSex field isn't present");
+        Assert.assertTrue(loginPage.isPrivacyPolicyCheckboxPresent(), "Privacy policy checkbox field isn't present");
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: you can use softAssert here

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 12]
```diff
@@ -85,4 +86,41 @@ public void testUIElements() {
...
+
+        softAssert.assertFalse(loginPage.isMaleSexChecked(), "MaleSex already checked");
+        softAssert.assertFalse(loginPage.isFemaleSexChecked(), "FemaleSex already checked");
+        Assert.assertFalse(loginPage.isPrivacyPolicyChecked(), "Privacy policy already checked");
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: the same

Comment (ashchavinska) [reply src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: done

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java (hunk 11 above)
Comment (ashchavinska) [reply src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: done

**Comment by vrudyk** [comment]
ok

**Pull request merged**